code/game/gamemodes/cult/cult.dm

```diff
@@ -7,8 +7,11 @@
 	var/anchor_time2kill = 5 MINUTES
 	var/bloodstone_cooldown = FALSE
 
-/proc/iscultist(mob/living/M)
-	return M?.mind?.has_antag_datum(/datum/antagonist/cult)
+/proc/iscultist(mob/living/M, require_cult = FALSE, holy_water_check = FALSE)
+	if(!istype(M))
+		return FALSE
+	var/datum/antagonist/cult/D = M?.mind?.has_antag_datum(/datum/antagonist/cult)
+	return D && (!require_cult || !D.agent) && (!holy_water_check || !D.ignore_holy_water)
```

Comment:
Im all for shortening procs whenever possible but this is stupidly long and unreadable
```suggestion
	if(!D) return FALSE
	if(require_cult && D.agent) return FALSE
	if(holy_water_check && D.ignore_holy_water) return FALSE
	return TRUE
```